File: tests/fixtures/sample.lua
local GraphWalker = {}
GraphWalker.__index = GraphWalker

function GraphWalker.buildMiniGraph(self, count)
    for _, n in ipairs(self.nodes) do
        print(n)
    end
end

local function lazyBuildContext(items)
    local gw = {nodes = items}
    setmetatable(gw, GraphWalker)
    return gw
end

local walker = lazyBuildContext({"a", "b"})
walker:buildMiniGraph(2)

File: tests/fixtures/sample.kt
class GraphWalker(private val nodes: List<String>) {
    fun buildMiniGraph(count: Int): Map<String, Any> {
        val result = mutableMapOf<String, Any>()
        for (n in nodes) {
            result[n] = count
        }
        return result
    }
}

fun lazyBuildContext(items: Array<String>): GraphWalker {
    return GraphWalker(items.toList())
}

fun main() {
    val walker = GraphWalker(listOf("a", "b"))
    val graph = walker.buildMiniGraph(2)
    println(graph)
}
